notebook/notebookapp.py

```diff
@@ -359,17 +359,18 @@ def parse_command_line(self, argv=None):
             self.port=int(self.extra_args[0])
 
     def start(self):
-        servers=list_running_servers(self.runtime_dir)
-        server=next((server for server in servers if server.get('port')==self.port),None)
-        if server: os.kill(server.get('pid'), signal.SIGTERM)
+        servers = list(list_running_servers(self.runtime_dir))
+        if not servers:
+            self.exit("There are no running servers")
+        for server in servers:
+            if server['port'] == self.port:
+                self.log.debug("Shutting down notebook server with PID: %i", server['pid'])
+                os.kill(server['pid'], signal.SIGTERM)
```

Comment:
I think you need a `break` in this branch so that it doesn't take the `else` below.